### PR TITLE
Upgrade workload using controller behind flag

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*

*Testing (if applicable):*

*Documentation added/planned (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/validations/cluster.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func clustersHaveEksaVersion(mgmtCluster, cluster *v1alpha1.Cluster) bool {
 		return false
 	}
 
-	if cluster.Spec.EksaVersion == nil && mgmtCluster.Spec.EksaVersion == nil {
+	if cluster.Spec.EksaVersion == nil || mgmtCluster.Spec.EksaVersion == nil {
```
Comment (g-gaston) [inline pkg/validations/cluster.go:180]: curious why this change? If it was a bug, could we add a test?

File: pkg/workflows/workload/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+		writer:           writer,
+		eksdInstaller:    eksdInstaller,
+		ClusterUpgrader:  clusterUpgrade,
+		packageInstaller: packageInstaller,
+	}
+}
+
+// Run Upgrade implements upgrade functionality for workload cluster's upgrade operation.
+func (c *Upgrade) Run(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec, validator interfaces.Validator) error {
+	logger.Info("POC New workflow creating workload cluster using the controller")
```
Comment (g-gaston) [inline pkg/workflows/workload/upgrade.go:47]: we already a similar log at the cmd level
if you wanna keep both just make sure we clean them up when removing the feature toggle

File: pkg/workflows/workload/upgradecluster.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type upgradeCluster struct{}
+
+// Run upgradeCluster performs actions needed to upgrade the management cluster.
+func (s *upgradeCluster) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Creating workload cluster")
```
Comment (g-gaston) [inline pkg/workflows/workload/upgradecluster.go:15]:
```suggestion
	logger.Info("Upgrading workload cluster components")
```

File: pkg/workflows/workload/validate.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/validations"
+)
+
+type setAndValidateWorkloadTask struct{}
+
+// SetAndValidateTask implementation
+
+// Run upgradeCluster performs actions needed to upgrade the workload cluster.
```
Comment (g-gaston) [inline pkg/workflows/workload/validate.go:16]: fix comments

File: pkg/workflows/workload/validate.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+func (s *setAndValidateWorkloadTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("POC Performing setup and validations")
```
Comment (g-gaston) [inline pkg/workflows/workload/validate.go:18]: now i'm just curious
why the same similar log everywhere? also tbh i'm a bit confused by the term poc
not big deal if you want to keep this while you test but it seems like extra work to clean later

File: pkg/workflows/workload/writeclusterconfig.go [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+	if err != nil {
+		commandContext.SetError(err)
+	}
+
+	successMsg := ""
+	if commandContext.CurrentClusterSpec != nil {
+		successMsg = "Cluster upgraded!"
+	} else {
+		successMsg = "Cluster created!"
+	}
```
Comment (g-gaston) [inline pkg/workflows/workload/writeclusterconfig.go:26]: the abstractions presented by `pkg/workflows` don't favor reusing tasks
the fact that you are tying to do this here seems to indicate maybe we should look into `pkg/workflow` instead. You don't need to do it, we can move forward with this PR as is. But maybe worth thinking about it

File: pkg/validations/cluster.go (hunk 1 above)
Comment (tatlat) [reply pkg/validations/cluster.go:180]: This change isn't needed. I just made it when I was looking into having the controller set eksaVersion.

**Review by g-gaston — COMMENTED** [review]
/lgtm

File: pkg/workflows/workload/writeclusterconfig.go (hunk 6 above)
Comment (tatlat) [reply pkg/workflows/workload/writeclusterconfig.go:26]: Yeah, I can fix that in a future PR.

**Pull request merged**